Forest loss driver prediction pipeline on OlmoEarth platform

favyen2 opened the pull request.

favyen2 committed:
Forest loss driver prediction pipeline on OlmoEarth platform.

favyen2 committed:
add pull-request to workflow so we can run it immediately?

favyen2 committed:
fix

favyen2 committed:
fix

favyen2 committed:
fix

favyen2 committed:
fix

favyen2 committed:
set STUDIO_API_KEY

favyen2 committed:
fix

uakfdotb committed:
fix

favyen2 committed:
add test for extract alerts

favyen2 committed:
Merge branch 'favyen/20251219-forest-loss-deploy' of github.com:allenai/olmoearth_projects into favyen/20251219-forest-loss-deploy

favyen2 committed:
add test for getting sentinel-2 assets

favyen2 committed:
Merge remote-tracking branch 'origin/main' into favyen/20251219-forest-loss-deploy

favyen2 committed:
update to new studio version which no longer calls it .geojson instead it has URL params

favyen2 committed:
fix imap_unordered issue

favyen2 committed:
Merge remote-tracking branch 'origin/main' into favyen/20251219-forest-loss-deploy

favyen2 committed:
Merge remote-tracking branch 'origin/main' into favyen/20251219-forest-loss-deploy

uakfdotb committed:
try with ubuntu-latest-m

favyen2 committed:
fix forest loss driver pipeline for studio api change

favyen2 committed:
Merge remote-tracking branch 'origin/main' into favyen/20251219-forest-loss-deploy

favyen2 committed:
use olmoearth_datasets

favyen2 committed:
set window success rate to 50%

favyen2 committed:
run in ecuador and bolivia

favyen2 committed:
Update forest loss driver model version.

favyen2 committed:
temporarily do a longer run (270 days)

favyen2 committed:
Change back to four months for future runs and make smaller tiles.

favyen2 committed:
Use points in the Studio jobs instead of the forest loss polygons.

favyen2 committed:
add polygon simplification and only send bounds to planetary computer search

favyen2 committed:
fix the max zoom at 12

favyen2 committed:
update beaker budget

favyen2 committed:
multi-period execution and switch to olmoearth_datasets data source

favyen2 committed:
add option to disable polygon smoothing

favyen2 committed:
don't rely on updated olmoearth_run in forest loss driver pipeline

instead we can just make the requests directly without using the pydantic models

APatrickJ commented on olmoearth_projects/projects/forest_loss_driver/deploy/__init__.py:
```diff
@@ -0,0 +1,661 @@
+            timeout=(REQUEST_TIMEOUT, UPLOAD_TIMEOUT),
+            headers=get_studio_headers(),
+        )
+        response.raise_for_status()
```
It might be helpful to extract this into a separate client, similar to what we did for the embeddings tutorial:

https://github.com/allenai/olmoearth_projects/blob/401d0b942dbb259d06a4fe28130c202e8f1a0cce/tutorials/embeddings/src/olmoearth_embeddings_tutorial/common/studio_client.py#L81-L97

APatrickJ replied:
FWIW this was an optional comment, just something to think about if we end up needing to reuse this code for something else later.

favyen2 replied:
agreed I think this is better, refactored it out to the client

APatrickJ commented on olmoearth_run_data/forest_loss_driver/deploy.Dockerfile:
```diff
@@ -0,0 +1,23 @@
+WORKDIR /olmoearth_projects
+
+COPY pyproject.toml /olmoearth_projects/pyproject.toml
+COPY uv.lock /rslearn/uv.lock
```
Should this be:

```
COPY uv.lock /olmoearth_projects/uv.lock
```

?

favyen2 replied:
fixed

APatrickJ commented on olmoearth_projects/projects/forest_loss_driver/deploy/make_tiles.py:
```diff
@@ -0,0 +1,103 @@
+
+        # Apply tippecanoe to convert the GeoJSON into a set of vector tiles.
+        local_tile_dir = os.path.join(tmp_dir, "tiles")
+        subprocess.call(
```
Maybe use `subprocess.check_call` here so you get an exception if the command fails. Or keep it as `call` but check the exit code.

favyen2 replied:
fixed, thanks!

APatrickJ commented on .github/workflows/forest_loss_driver_prediction.yaml:
```diff
@@ -0,0 +1,74 @@
+name: Forest-Loss-Driver-Prediction
+
+on:
+  pull_request:
```
Probably want to configure `paths` so this doesn't run on every pull request, but only pull requests that touch related directories. Something like:

```
    paths: ['olmoearth_projects/projects/forest_loss_driver/**', 'olmoearth_run_data/forest_loss_driver/**']
```

favyen2 replied:
ah right I forgot about that. I will keep it like that for now so I can run it again tomorrow for the weekly inference, but will keep this comment open so I remember to remove it before merging the PR.

favyen2 replied:
oh just to be clear I meant that I want it to just run weekly when it's merged, we don't need that pull_request trigger at all, it's only there so I can manually run the workflow for now since otherwise the option only shows up based on workflows that are present on main branch.

APatrickJ replied:
Okay got it. I think you only need `workflow_dispatch` to manually trigger, not `pull_request`

favyen2 replied:
Yeah I think that was the problem, with workflow_dispatch it only shows up if you have it merged into main branch but this PR isn't merged yet.

APatrickJ commented on olmoearth_projects/projects/forest_loss_driver/deploy/__init__.py:
```diff
@@ -0,0 +1,661 @@
+            continue
+        elif (
+            job_status not in ["completed", "failed"]
+        ):  # tmp: we need to treat failed as completed since some jobs are stuck failed but have outputs
```
This is a weird one! I'm curious how often that happens / whether it's still an issue in the platform. Maybe log when we're proceeding anyway with a failed status?

APatrickJ replied:
I'm still curious if you're still seeing this or if this is more of an artifact of an earlier bug. I think it'd be great to track this somehow because if you're still seeing it then it sounds like a bug we should fix.

favyen2 replied:
I think this isn't needed anymore -- removed and will see if it works tomorrow with new run.

APatrickJ commented on olmoearth_projects/projects/forest_loss_driver/deploy/sentinel2.py:
```diff
@@ -0,0 +1,261 @@
+        kwargs_list: list of kwargs dicts to pass to the function.
+
+    Returns:
+        generator for outputs from the function in arbitrary order.
```
Seems fine to document arbitrary order though I think `imap` preserves submission order. I wonder if you wanted `imap_unordered`?

favyen2 replied:
Fixed the docstring, it must be from copying and pasting from `rslearn.utils.mp` (pre-AI-agent). It should be imap since we use it with the `zip(features, outputs)`.

favyen2 committed:
use check_call instead of call

favyen2 committed:
Merge remote-tracking branch 'origin/main' into favyen/20251219-forest-loss-deploy

favyen2 committed:
Fix uv.lock copy in deploy.Dockerfile

favyen2 committed:
Fix broken test needing secret, and fix star_map docstring.

APatrickJ commented on olmoearth_projects/projects/forest_loss_driver/deploy/__init__.py:
```diff
@@ -0,0 +1,661 @@
+
+def _get_most_recent_friday() -> datetime:
+    """Get the most recent Friday."""
+    now = datetime.now()
```
Should we use UTC or specify a timezone here?

favyen2 replied:
hm I vaguely remember something about it but I can't remember, seems like it should be fine to use UTC

favyen2 committed:
refactor basic studio API calls to separate utility class

favyen2 committed:
use UTC when getting most recent Friday

favyen2 committed:
remove pull request trigger since we are about to merge it

after merging, workflow_dispatch will work (should be able to manually
trigger from Actions once the workflow with workflow_dispatch is on
whatever branch is marked as the primary branch in the repository)
